dags/openshift_nightlies/dag.py

```diff
@@ -77,86 +63,78 @@ def _get_openshift_installer(self):
         raise NotImplementedError()
 
     def _get_e2e_benchmarks(self): 
-        return e2e.E2EBenchmarks(self.dag, self.version, self.release_stream, self.platform, self.profile, self.metadata_args)
+        return e2e.E2EBenchmarks(self.dag, self.release)
 
     def _get_scale_ci_diagnosis(self):
-        return scale_ci_diagnosis.Diagnosis(self.dag, self.version, self.release_stream, self.latest_release, self.platform, self.profile, self.metadata_args)
+        return scale_ci_diagnosis.Diagnosis(self.dag, self.release)
 
 
 class CloudOpenshiftNightlyDAG(AbstractOpenshiftNightlyDAG):
-    def __init__(self, version, release_stream, platform, profile, version_alias):
-        super().__init__(version, release_stream, platform, profile, version_alias)
-        self.release_stream_base_url = Variable.get("release_stream_base_url")
-        self.latest_release = var_loader.get_latest_release_from_stream(self.release_stream_base_url, self.release_stream)
-    
     def build(self):
         installer = self._get_openshift_installer()
         install_cluster = installer.get_install_task()
-        cleanup_cluster = installer.get_cleanup_task()
+        
         with TaskGroup("utils", prefix_group_id=False, dag=self.dag) as utils:
             utils_tasks=self._get_scale_ci_diagnosis().get_utils()
             chain(*utils_tasks)
-            utils_tasks[-1] >> cleanup_cluster
+
         with TaskGroup("benchmarks", prefix_group_id=False, dag=self.dag) as benchmarks:
             benchmark_tasks = self._get_e2e_benchmarks().get_benchmarks()
             chain(*benchmark_tasks)
-            benchmark_tasks[-1] >> utils
-
-        install_cluster >> benchmarks
+            
+        if self.config.cleanup_on_success:
+            cleanup_cluster = installer.get_cleanup_task()
+            install_cluster >> benchmarks >> utils >> cleanup_cluster
```

Comment:
Is this some kind of ordering you specifying ?

Reply:
yeah this is actually drawing the dag dependencies and stuff. Essentially this is saying "cleanup_cluster depends on utils which depends on the benchmarks which depend on install_cluster"

Reply:
Can you please add that as a comment in the code itself? Could make things more readable.